src/com/google/enterprise/adaptor/opentext/OpentextAdaptor.java

```diff
@@ -418,6 +426,106 @@ public void init(AdaptorContext context) {
       }
     }
 
+    // This URL is used for search and download.
+    this.indexingContentServerUrl =
+        config.getValue("opentext.indexing.contentServerUrl");
+    if (this.indexingContentServerUrl.isEmpty()) {
+      this.indexingContentServerUrl = this.contentServerUrl;
+    }
+    log.log(Level.CONFIG, "opentext.indexing.contentServerUrl: {0}",
+        this.indexingContentServerUrl);
+
+    try {
+      this.readTimeout = Integer.parseInt(
+          config.getValue("adaptor.docContentTimeoutSecs")) * 1000;
+    } catch (NumberFormatException e) {
+      throw new InvalidConfigurationException(
+          "Invalid value for adaptor.docContentTimeoutSecs: "
+          + config.getValue("adaptor.docContentTimeoutSecs"));
+    }
+    try {
+      this.connectTimeout = Integer.parseInt(
+          config.getValue("adaptor.docHeaderTimeoutSecs")) * 1000;
+    } catch (NumberFormatException e) {
+      throw new InvalidConfigurationException(
+          "Invalid value for adaptor.docHeaderTimeoutSecs: "
+          + config.getValue("adaptor.docHeaderTimeoutSecs"));
+    }
+
+    // Don't log these parameters unless they're set to something
+    // other than the default.
+    this.downloadMethod = config.getValue("opentext.indexing.downloadMethod");
+    if (!this.downloadMethod.equals("contentserver")) {
+      if (!(this.downloadMethod.equals("cws")
+              || this.downloadMethod.equals("cwscontenthandler"))) {
+        throw new InvalidConfigurationException(
+            "Invalid value for opentext.indexing.downloadMethod");
```

Comment:
Include the value, maybe include the legal values. Might consider an enum if that's handy elsewhere at all.

Guava's Enums.getIfPresent(DownloadMethod.class, string.toUpperCase(US)).or(DownloadMethod.CONTENTHANDLER) could be handy, as is
"Valid values: " + Arrays.toString(DownloadMethod.values()).toLowerCase(US)